launch/test/launch/substitutions/test_equals_substitution.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the EqualsSubstitution class."""
+
+from launch import LaunchContext
+
+from launch.substitutions import EqualsSubstitution
+from launch.substitutions import PathJoinSubstitution
+
+import os
+
+
+def test_equals_substitution():
+    def _permute_assertion(left, right, context, output):
+        assert EqualsSubstitution(left, right).perform(context) == output
+        assert EqualsSubstitution(right, left).perform(context) == output
+
+    lc = LaunchContext()
+
+    # NoneType
+    _permute_assertion(None, None, lc, 'true')
+    _permute_assertion(None, "", lc, 'true')
+    _permute_assertion(None, "something", lc, 'false')
```

Comment:
style nit: avoid double-quotes if possible and use single quotes everywhere. I'm surprised our linter is happy with this :thinking: